src/dotnet/commands/dotnet-run/RunCommand.cs

```diff
@@ -133,21 +133,22 @@ private int RunExecutable()
             }
 
             // Locate the runtime
-            string runtime = Environment.GetEnvironmentVariable("DOTNET_HOME");
-            if (string.IsNullOrEmpty(runtime))
+            string dotnetHome = Environment.GetEnvironmentVariable("DOTNET_HOME");
+            if (string.IsNullOrEmpty(dotnetHome))
             {
                 // Use the runtime deployed with the tools, if present
-                var candidate = Path.Combine(AppContext.BaseDirectory, "..", "runtime");
+                var candidateBase = Path.Combine(AppContext.BaseDirectory, "..");
+                var candidate = Path.Combine(candidateBase, "runtime");
                 if (File.Exists(Path.Combine(candidate, Constants.LibCoreClrName)))
```

Comment:
`libcoreclr` is actually in `DOTNET_HOME/runtime/coreclr`, not `DOTNET_HOME/runtime`